client/lib/signup/cart.js

```diff
@@ -11,7 +11,33 @@ var wpcom = require( 'lib/wp' ),
 	cartValues = require( 'lib/cart-values' ),
 	cartItems = cartValues.cartItems;
 
+function addProductsToCart( cart, newCartItems ) {
+	forEach( newCartItems, function( cartItem ) {
+		cartItem.extra = Object.assign( cartItem.extra || {}, {
+			context: 'signup'
+		} );
+		const addFunction = cartItems.add( cartItem );
+
+		cart = cartValues.fillInAllCartItemAttributes( addFunction( cart ), productsList.get() );
+	} );
+
+	return cart;
+}
+
 module.exports = {
+	createCart: function( cartKey, newCartItems, callback ) {
+		let newCart = {
+			cart_key: cartKey,
+			products: [],
+			temporary: false,
+		};
+
+		newCart = addProductsToCart( newCart, newCartItems );
+
+		wpcom.undocumented().cart( cartKey, 'POST', newCart, function( postError ) {
```

Comment:
Maybe you wanted to make `callback` optional and missed the check here if `callback` is a function?